### PR TITLE
missing juttle-process tests

### LINKED ISSUE #98: Code coverage ignores juttle-subprocess
https://github.com/juttle/outrigger/commit/9bc435832f0091cc29439c5c7c767138993a7fb7#diff-b9e12334e9eafd8341a6107dd98510c9R57 is not that great.

### PR DESCRIPTION
**Pull request by rlgomes**
fixes #98

we were missing tests here that didn't depend on tracking the coverage
across spawn process which would be much harder to do.

### CODE REVIEW

**Comment by rlgomes** [comment]
@mstemm @davidgalbraith @go-oleg @mnibecker  missing juttle-process tests. Want to get your opinion on the approach before I expand things out to fully cover that one file and the various message types being sent back and forth.

File: test/lib/juttle-subprocess.spec.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                throw new Error('Message not found with fields ' +
+                                JSON.stringify(fields) + ' in ' +
+                                JSON.stringify(sentData));
+            }
+        }, { max_tries: 10, interval: 100 });
+    };
+
+    beforeEach(function() {
+        oldArgv = process.argv;
+        process.argv = ['node', 'juttle-subproces.js', ''];
```
Comment (mstemm) [inline test/lib/juttle-subprocess.spec.js]: This is a typo right?

**Comment by mstemm** [comment]
This approach looks fine to me--having tests at this level make sense to me.

Brainstorming on addl tests to add:
 - testing that all messages describing sinks, data, etc are sent for a good program.
 - test passing the juttle config location to the subprocess via argv
 - ensure that compile, juttle runtime, etc errors have a location in the error message

File: test/lib/juttle-subprocess.spec.js (hunk 1 above)
Comment (rlgomes) [inline test/lib/juttle-subprocess.spec.js]: Ah yes, this little trick is to make the subprocess happy when grabbing the `argv[2]` so the typo "doesn't matter" but I'll correct it.

**Comment by rlgomes** [comment]
@mstemm I'll add those other tests  in this PR as well.

**Pull request merged**